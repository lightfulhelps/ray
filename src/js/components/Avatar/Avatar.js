// @flow
import React from 'react';
import classNames from 'classnames';
import { Icon } from '../../';

type Props = {
  className?: string,
  provider?: 'facebook' | 'twitter' | 'linkedin',
  url?: string,
};

const Avatar = ({ className, url, provider }: Props) => {
  const classes = classNames(className, 'avatar');

  return (
    <div className={classes}>
      <div className="avatar-image-wrap">
        {url && <img className="avatar-image" src={url} alt="" />}
      </div>
      {provider && (
        <div className={`avatar-provider avatar-provider-${provider}`}>
          <Icon name={provider} color="#ffffff" size={16} />
        </div>
      )}
    </div>
  );
};

export default Avatar;
