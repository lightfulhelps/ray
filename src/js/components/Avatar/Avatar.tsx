
import React from "react";
import classNames from "classnames";
import { Icon } from "../../";

type Props = {
  className?: string;
  isDisconnected?: boolean;
  provider?: "facebook" | "twitter" | "linkedin";
  providerSize?: "sm" | "md" | "lg";
  url?: string;
};

const Avatar = ({
  className,
  isDisconnected,
  url = '',
  provider,
  providerSize = 'md',
  ...other
}: Props) => {
  const classes = classNames(className, 'avatar', `avatar-provider-${providerSize}`, {
    'avatar-disconnected': isDisconnected
  });

  return <div {...other} className={classes}>
      <div className="avatar-inner">
        <div className="avatar-image" style={{ backgroundImage: `url(${url})` }} />
        {provider && <div className={`avatar-provider avatar-provider-${provider}`}>
            <Icon name={provider} color="#ffffff" />
          </div>}
      </div>
    </div>;
};

export default Avatar;