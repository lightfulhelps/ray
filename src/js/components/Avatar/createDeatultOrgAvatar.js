const createDeatultOrgAvatar = orgName => {
  const letter = orgName ? orgName[0] : ' ';
  let bgClass;
  if (/[A-F]/.test(letter)) {
    bgClass = 'bg-secondary';
  }
  if (/[G-L]/.test(letter)) {
    bgClass = 'bg-primary';
  }
  if (/[M-R]/.test(letter)) {
    bgClass = 'bg-danger';
  }
  if (/[S-Z]/.test(letter)) {
    bgClass = 'bg-warning';
  }
  if (!/[A-Z]/.test(letter)) {
    bgClass = 'bg-info';
  }
  return { bgClass, letter };
};

export default createDeatultOrgAvatar;
