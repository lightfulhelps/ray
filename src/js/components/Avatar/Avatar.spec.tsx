import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Avatar from './Avatar';
import { Icon } from '../..';

const setup = (overrides = {}) => {
  const props = merge(
    {
      url: 'https://randomuser.me/api/portraits/women/47.jpg',
      name: 'Milton Friesen',
      provider: 'twitter' as const,
    },
    overrides
  );
  const wrapper = shallow(<Avatar {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Avatar />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup({ className: 'custom' });

    expect(wrapper.hasClass('avatar')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the url prop', () => {
    const { wrapper } = setup({ url: 'https://randomuser.me/api/portraits/women/47.jpg' });

    expect(wrapper.find('.avatar-image').prop('style').backgroundImage).toContain(
      'https://randomuser.me/api/portraits/women/47.jpg'
    );
  });

  it('should handle the provider prop', () => {
    const { wrapper } = setup({ provider: null });

    expect(wrapper.find('.avatar-provider').exists()).toBe(false);

    wrapper.setProps({ provider: 'facebook' });

    expect(wrapper.find('.avatar-provider').exists()).toBe(true);
    expect(
      wrapper
        .find('.avatar-provider')
        .find(Icon)
        .prop('name')
    ).toEqual('facebook');
  });

  it('should handle the providerSize prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ providerSize: 'lg' });

    expect(wrapper.hasClass('avatar-provider-lg')).toBe(true);

    wrapper.setProps({ providerSize: 'sm' });

    expect(wrapper.hasClass('avatar-provider-sm')).toBe(true);
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });

  it('should display an initials avatar if no image url', () => {
    const { wrapper } = setup({ url: null });

    expect(wrapper.find('.avatar-image').prop('style').backgroundImage).toContain(
      'url(data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20style%3D%22isolation%3Aisolate%3B%22%20viewBox%3D%220%200%201%201%22%20version%3D%221.1%22%3E%3Crect%20width%3D%221%22%20height%3D%221%22%20fill%3D%22transparent%22%2F%3E%3Ctext%20x%3D%2250%25%22%20y%3D%2250%25%22%20style%3D%22%20font-family%3A%20Arial%2Csans-serif%3B%20font-size%3A%200.5px%22%20fill%3D%22%23FFF%22%20text-anchor%3D%22middle%22%20dy%3D%22.178%22%3EMF%3C%2Ftext%3E%3C%2Fsvg%3E), url(https://assets.lightful.com/platform/brand/avatar_background.png)'
    );
  });
});
