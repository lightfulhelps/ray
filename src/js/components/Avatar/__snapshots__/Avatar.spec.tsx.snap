// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render 1`] = `
<div
  className="avatar avatar-provider-md"
>
  <div
    className="avatar-inner"
  >
    <div
      className="avatar-image"
      style={
        Object {
          "backgroundImage": "url(https://randomuser.me/api/portraits/women/47.jpg)",
        }
      }
    />
    <div
      className="avatar-provider avatar-provider-twitter"
    >
      <Icon
        color="#ffffff"
        name="twitter"
      />
    </div>
  </div>
</div>
`;
